### PR TITLE
feat: add Windows and macOS support

### PR DESCRIPTION
**Pull request by kaiguo**
react-native-gesture-handler doesn't have Windows/macOS implementation yet, use the web version as stubs for now to enable navigation features for Windows and mac.

Close https://github.com/microsoft/react-native-windows/issues/3884
Close https://github.com/microsoft/react-native-windows/issues/4044

### CODE REVIEW

**Comment by elcssmouhsine** [comment]
could you also add react-native-macos support while at it :

    animationEnabled: Platform.OS !== 'web' && Platform.OS !== 'windows' && Platform.OS !== 'macos',

**Comment by kaiguo** [comment]
> could you also add react-native-macos support while at it

Yep, added : )

**Comment by elcssmouhsine** [comment]
react-native-gesture-handler does not support macos

[issue #1052](https://github.com/software-mansion/react-native-gesture-handler/issues/1052)

File: packages/drawer/src/views/Drawer.tsx [hunk 1]
```diff
@@ -489,7 +495,9 @@ export default class DrawerView extends React.Component<Props> {
     // Check if the drawer width is available to avoid division by zero
     eq(this.drawerWidth, 0),
     0,
-    abs(divide(this.translateX, this.drawerWidth))
+    Platform.OS === 'windows' || Platform.OS === 'macos'
```
Comment (satya164) [inline packages/drawer/src/views/Drawer.tsx:498]: Maybe you could add a comment about why this is needed

File: packages/drawer/src/views/GestureHandlerCompat.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
```
Comment (satya164) [inline packages/drawer/src/views/GestureHandlerCompat.tsx:1]: Maybe this file could be named as `GestureHandlerStub` which would be more appropriate

File: packages/stack/src/views/GestureHandlerCompat.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import * as React from 'react';
```
Comment (satya164) [inline packages/stack/src/views/GestureHandlerCompat.tsx:1]: Maybe this file could be named as `GestureHandlerStub` which would be more appropriate

File: packages/stack/src/views/MaskedView.macos.tsx [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import StubbedMaskedView from './MaskedView';
```
Comment (satya164) [inline packages/stack/src/views/MaskedView.macos.tsx:1]: Wouldn't this resolve to `MaskedView.native.tsx`? Maybe we need to rename `MaskedView.tsx` to `MaskedViewStub.tsx` and then re-export it.

```js
export * from './MaskedViewStub';
```

File: packages/drawer/src/views/Drawer.tsx (hunk 1 above)
Comment (kaiguo) [reply packages/drawer/src/views/Drawer.tsx]: Moved the platform check to Overlay.tsx and added some comments.

**Comment by kaiguo** [comment]
Thanks @satya164! Just pushed out another commit to address your feedbacks.

**Comment by kaiguo** [comment]
Hey @satya164 do we know when the next release will be? Looks like the current release doesn't have this change and your refactor yet.

**Comment by satya164** [comment]
I need to finish some other stuff before doing a release

**Comment by Zorgatone** [comment]
Waiting for the release as well

**Comment by lh9403** [comment]
Is MacOS supported now?

**Pull request merged**